packages/core-flows/src/product/workflows/update-products.ts

```diff
@@ -15,6 +15,7 @@ export const updateProductsWorkflow = createWorkflow(
   (
     input: WorkflowData<WorkflowInput>
   ): WorkflowData<ProductTypes.ProductDTO[]> => {
+    // TODO: Delete price sets for removed variants
```

Comment:
question: Wouldn't the cascade on the link definition take care of this? At least I would expect it to.

Reply:
Theoretically, when calling remoteLink.delete, it will call the module service softDelete method, to cascade the deletion, the soft delete will return the deleted entries deeply mapped with the linkable keys and allow to perform the cascade soft delete to all the module services

Reply:
I think the only issue is that we currently don't do a soft-delete in the upsertWithReplace, and at this point we don't handle the deleted variants in any way. In the `deleteVariants` workflow this is already handled. Since this doesn't block end-to-end flow for now I decided not to do it for now.